official/resnet/keras/keras_common.py

```diff
@@ -0,0 +1,191 @@
+      self.record_batch = True
+      # TODO(anjalisridhar): add timestamp as well.
+      if batch != 0:
+        tf.logging.info("BenchmarkMetric: {'num_batches':%d, 'time_taken': %f,"
+                        "'images_per_second': %f}" %
+                        (batch, last_n_batches, examples_per_second))
+
+class LearningRateBatchScheduler(tf.keras.callbacks.Callback):
+  """Callback to update learning rate on every batch (not epoch boundaries).
+
+  N.B. Only support Keras optimizers, not TF optimizers.
+
+  Args:
+      schedule: a function that takes an epoch index and a batch index as input
+          (both integer, indexed from 0) and returns a new learning rate as
+          output (float).
+  """
+
+  def __init__(self, schedule, batch_size, num_images):
+    super(LearningRateBatchScheduler, self).__init__()
+    self.schedule = schedule
+    self.batches_per_epoch = num_images / batch_size
+    self.batch_size = batch_size
+    self.epochs = -1
+    self.prev_lr = -1
+
+  def on_epoch_begin(self, epoch, logs=None):
+    if not hasattr(self.model.optimizer, 'learning_rate'):
+      raise ValueError('Optimizer must have a "learning_rate" attribute.')
+    self.epochs += 1
+
+  def on_batch_begin(self, batch, logs=None):
+    lr = self.schedule(self.epochs, batch, self.batches_per_epoch, self.batch_size)
+    if not isinstance(lr, (float, np.float32, np.float64)):
+      raise ValueError('The output of the "schedule" function should be float.')
+    if lr != self.prev_lr:
+      self.model.optimizer.learning_rate = lr  # lr should be a float here
+      # tf.keras.backend.set_value(self.model.optimizer.learning_rate, lr)
+      self.prev_lr = lr
+      tf.logging.debug('Epoch %05d Batch %05d: LearningRateBatchScheduler change '
+                   'learning rate to %s.', self.epochs, batch, lr)
+
+def get_optimizer():
+  if FLAGS.use_tf_momentum_optimizer:
+    learning_rate = BASE_LEARNING_RATE * FLAGS.batch_size / 256
+    optimizer = tf.train.MomentumOptimizer(learning_rate=learning_rate, momentum=0.9)
+  else:
+    # optimizer = tf.keras.optimizers.SGD(learning_rate=0.1, momentum=0.9)
+    optimizer = gradient_descent_v2.SGD(learning_rate=0.1, momentum=0.9)
+
+  return optimizer
+
+
+def get_callbacks(learning_rate_schedule_fn, num_images):
+  time_callback = TimeHistory(FLAGS.batch_size)
+
+  tensorboard_callback = tf.keras.callbacks.TensorBoard(
+    log_dir=FLAGS.model_dir)
+
+  lr_callback = LearningRateBatchScheduler(
+    learning_rate_schedule_fn,
+    batch_size=FLAGS.batch_size,
+    num_images=num_images)
+
+  return time_callback, tensorboard_callback, lr_callback
+
+def analyze_fit_and_eval_result(history, eval_output):
+  stats = {}
+  stats['accuracy_top_1'] = eval_output[1]
+  stats['eval_loss'] = eval_output[0]
+  stats['training_loss'] = history.history['loss'][-1]
+  stats['training_accuracy_top_1'] = history.history['categorical_accuracy'][-1]
+
+  print('Test loss:{}'.format(stats['eval_loss']))
```

Comment:
You also could just remove this if you wanted.  This may have been a print statement I left in the code.  Keras already prints these numbers.  If I printed something I would format the entire (or mostly) stats dict nicely and print that out with user friendly names.

Reply:
Removed all.